Reserveringssysteem/CancelReservation.cs

```diff
@@ -6,68 +6,88 @@
+                        if (ReservationsList[i].ReservationId.ToLower() == input.ToLower())     //Als een match gevonden wordt...
                         {
                             Console.Clear();
                             CancelTitle();
-                            Console.WriteLine("De code staat in het systeem en de reservering wordt nu geannuleerd..");
-                            foundItem = true ;
-                            ReservationsList.Remove(ReservationsList[i]);                      // Verwijderd bijbehorende item uit JSON
-                            Serialize(ReservationsList,"reservations.json");                   // Slaat de JSON opnieuw op na de aanpassing
+                            AreUSure();                                                         // Laat het keuzemenu zien; ja voor verwijderen <--> nee voor terug naar vorige scherm
+                            keepTrue = false;
                             break;
                         }
                     }
-                    }
-                if (foundItem == false)                                                // Als er geen match is vragen of je het opnieuw wilt proberen
-                {
-                    Console.Clear();
-                    CancelTitle();
-                    Console.WriteLine("\nReserveringscode niet herkend.. Probeer opnieuw of ga terug naar het hoofdmenu door op 'enter' te drukken: ");
-                    input = Console.ReadLine();
-                    if (input == "")
-                        break;
                 }
-                else if (foundItem)                                                     // Sluit af als reservering geannuleerd is
+                else if (input == "")
                 {
-                    Console.WriteLine("\nDe reservering is geannuleerd!");
+                    keepTrue = false;
                     Utils.Enter();
-                    break;
+                }
+                else if (input.Length != 4 && input != "")
+                {
+                    input = InputAgain();
                 }
             }
         }
-
+        static void AreUSure() => SelectionMenu.Make(new string[2] { "Ja", "Nee" }, actions: new Action[] { ja, nee }, Logo.Annuleren, "\nWeet u zeker dat u de reservering wilt annuleren?\n");
+        public static void ja()
+        {
+            DeleteReservation();
+        }
+        private static void DeleteReservation()
+        {
+            ReservationsList = Deserialize<List<Reservation>>("reservations.json");
+            ReservationsList.Remove(ReservationsList[i]);
+            Console.Clear();
+            CancelTitle();
+            Serialize(ReservationsList, "reservations.json");                   // Slaat de JSON opnieuw op na de aanpassing
+            Console.WriteLine("\nDe reservering is verwijderd.\nAls u opnieuw wilt reserveren kunt u dat doen als u op 'enter' drukt.");
```

Comment:
Not sure, maar volgens mij ga je niet direct terug naar het reserveer gedeelte, dus zou ik zeggen i.p.v. "opnieuw reserveren.." zeggen "Om terug te gaan naar het hoofdmenu". Iets op die fiets.